app/routes.py

```diff
@@ -87,10 +93,21 @@ def logout():
 # =========================
 # 2. Login Required
 # # =========================
+@app.route('/bookmark/<book_id>', methods=['GET', 'POST'])

```

Comment:
In general, "creating resources" should always be done using POST request. The way you implemented are only using GET requests, which I guess is fine for now.

Please remove 'POST'  from the allowed methods